test/lib/godmin/helpers/filters_test.rb

```diff
@@ -13,13 +13,36 @@ def test_filter_form_is_a_bootstrap_form_builder
         end
       end
 
-      # TODO: actually implement this test properly, apparently
-      # assert_select can be used if the helper output is parsed
-      # with HTML::Document first...
-      def test_string_filter_field
-        form = filter_form url: "/" do |f|
+      def test_string_renders_as_string
+        @output_buffer = filter_form url: "/" do |f|
           f.filter_field :foo, as: :string
         end
+
+        assert_select "input[type=text][name=?]", "filter[foo]", 1, "No text input found with name foo"
+      end
+
+      def test_string_filter_field_renders_text_field
+        @output_buffer = filter_form url: "/" do |f|
+          f.string_filter_field :foo, {}, value: "Foobar"
+        end
+
+        assert_select "label[for=foo]", 1, "No label was found"
+        assert_select "input[type=text][name=?]", "filter[foo]", 1, "No text input found with name foo" do |element|
+          assert_equal "Foo", element.attr("placeholder").text
+          assert_equal "Foobar", element.attr("value").text
+        end
+      end
+
+      def test_multiselect_filter_field_includes_no_hidden
```

Comment:
Should we split this test in two? One general test for the multiselect filter, like the one for the string filter, and one for not including a hidden field?